### ISSUE TITLE
`let_unit_value` suggestion replacing variable in println with unit

### ISSUE THREAD

**Issue by MolotovCherry** [rust-lang/rust-clippy#15494]
### Summary

`let_unit_value` suggestion is turning a statement like `println!("{a:?});` into `println!("{():?});`

### Lint Name

let_unit_value

### Reproducer

I tried this code:

```rust
let a = vec![1,2,3].resize(0, 5);
println!("{a:?}");
```

I saw this happen:

```
warning: this let-binding has unit value
 --> src/main.rs:4:1
  |
4 | let a = vec![1,2,3].resize(0, 5);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
  = note: `#[warn(clippy::let_unit_value)]` on by default
help: omit the `let` binding and replace variable usages with `()`
  |
4 + vec![1,2,3].resize(0, 5);
5 ~ println!("{():?}");
  |
```

I expected to see this happen (no suggestion):
```
warning: this let-binding has unit value
 --> src/main.rs:4:1
  |
4 | let a = vec![1,2,3].resize(0, 5);
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
  = note: `#[warn(clippy::let_unit_value)]` on by default
```

### Version

```text
1.89 stable rustc

0.1.89 (2025-08-04 29483883ee)
```

### Additional Labels

@rustbot label: +I-suggestion-causes-error

**Comment by ada4a** [comment]
duplicate of https://github.com/rust-lang/rust-clippy/issues/15061, fixed in https://github.com/rust-lang/rust-clippy/pull/15085

@rustbot label S-fixed

**Comment by profetia** [comment]
Closed in #15085
